StRoot/StPicoDstMaker/StPicoDstMaker.cxx

```diff
@@ -1137,10 +1137,15 @@ void StPicoDstMaker::fillTracks() {
       picoTrk->setDedxError( gTrk->probPidTraits().dEdxErrorFit() );
     }
 
-#if defined (__TFG__VERSION__)
     picoTrk->setDndx( gTrk->probPidTraits().dNdxFit() );
     picoTrk->setDndxError( gTrk->probPidTraits().dNdxErrorFit() );
-#endif /* __TFG__VERSION__ */
+    if ( gTrk->primaryTrack() ) {
+      picoTrk->setStatus(1);
+    }
+    else {
+      picoTrk->setStatus(0);
+    }
```

Comment:
It looks as if this new `mStatus` member is added to distinguish between primary and global tracks. But we already have `StPicoTrack::IsPrimary()`. Wouldn't it be confusing to have two flags with the same meaning?

Reply:
Grigory, is this PR you mentioned today to resolve for the new library tagging?
Dmitri, my understanding is that mStatus will be set to 1 if this track is attached to any primary vertex (StMuTrack::primaryTrack()). While StPicoTrack::IsPrimary() is true only when the track is associate to the primary vertex stored in picoDst.

Reply:
Xin, if I interpret your statement correctly you are saying that the definition of primary tracks changes in PicoDst w.r.t. MuDst, i.e. some tracks defined as primary in MuDst may not be primary in PicoDst.

I agree with the first part of your statement about tracks defined primary if they associate with any vertex but I can't confirm the second half about the StPicoTrack::IsPrimary(). Looking at the [loop over global MuDst tracks](https://github.com/star-bnl/star-sw/blob/c71b0a035d7b3b31cad4191fe54c7acac1b7e8ac/StRoot/StPicoDstMaker/StPicoDstMaker.cxx#L1030-L1578) where the primary momentum of StPicoTrack is set it is not apparent to me how the definition of the primary tracks changes... It is possible that I missed the logic in this 500+line loop but a reference to the code would help in this case.

Reply:
Dmitri, I think this is correct. In MuDst, we store multiple vertices, and tracks associated to any primary vertex are considered primary tracks, though some may not be the primary for the current MuVertex. In PicoDst, we only store one primary vertex (best one for analysis) and then only primary tracks associated with this vertex are flagged as primary in PicoDst. From the analysis point of view, often for HI data, the vertex selection is pretty settled while primary vertex selection in high luminosity / small system collisions (pp or pA), the primary vertex/track selection may need some special studies. I think this may be one reason why many spin analysis still uses MuDst data.

I remember Grigory was thinking also to add multiple vertices into PicoDst too. I think this may be some future development.

Reply:
Hi Xin, I am fine with the logic you describe and the difference between the Pico and MuDst primary track.  My concern was that I don't quite see what you say implemented in the pico code... Anyway, I leave it to you, Grigory and Rongrong to sort out. No further comments from my side.